internal/kv/kv.go

```diff
@@ -0,0 +1,50 @@
+package kv
+
+import (
+	"context"
+	"crypto/x509"
+
+	"github.com/ydb-platform/ydb-go-genproto/draft/Ydb_KeyValue_V1"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/metadata"
+
+	"github.com/ydb-platform/terraform-provider-ydb/sdk/terraform/auth"
+)
+
+type ClientParams struct {
+	DatabaseEndpoint string
+	Database         string
+	UseTLS           bool
+	AuthCreds        auth.YdbCredentials
+}
+
+func CreateDBConnection(ctx context.Context, params ClientParams) (*grpc.ClientConn, error) {
+	var opts grpc.DialOption
+
+	switch params.UseTLS {
+	case true:
+		pool, _ := x509.SystemCertPool()
+		creds := credentials.NewClientTLSFromCert(pool, "")
+		opts = grpc.WithTransportCredentials(creds)
+	case false:
+		opts = grpc.WithTransportCredentials(insecure.NewCredentials())
+	}
```

Comment:
Probably, better could be

```go
if params.UseTLS {
    // some stuff
} else {

}
```

switch for boolean variable is overkill, probably

Reply:
done